cpp/src/strings/contains.cu

```diff
@@ -27,65 +25,61 @@
 #include <cudf/strings/contains.hpp>
 #include <cudf/strings/detail/utilities.hpp>
 #include <cudf/strings/string_view.cuh>
-#include <cudf/strings/strings_column_view.hpp>
 
 #include <rmm/cuda_stream_view.hpp>
-#include <rmm/exec_policy.hpp>
-
-#include <thrust/iterator/counting_iterator.h>
-#include <thrust/transform.h>
 
 namespace cudf {
 namespace strings {
 namespace detail {
 
 namespace {
 /**
- * @brief This functor handles both contains_re and match_re to minimize the number
- * of regex calls to find() to be inlined greatly reducing compile time.
+ * @brief This functor handles both contains_re and match_re to regex-match a pattern
+ * to each string in a column.
  */
-template <int stack_size>
 struct contains_fn {
-  reprog_device prog;
   column_device_view const d_strings;
-  bool const beginning_only;  // do not make this a template parameter to keep compile times down
+  bool const beginning_only;
 
-  __device__ bool operator()(size_type idx)
+  __device__ bool operator()(size_type const idx,
+                             reprog_device const prog,
+                             int32_t const thread_idx)
   {
     if (d_strings.is_null(idx)) return false;
     auto const d_str = d_strings.element<string_view>(idx);
-    int32_t begin    = 0;
-    int32_t end      = beginning_only ? 1    // match only the beginning of the string;
-                                      : -1;  // match anywhere in the string
-    return static_cast<bool>(prog.find<stack_size>(idx, d_str, begin, end));
+
+    size_type begin = 0;
+    size_type end   = beginning_only ? 1    // match only the beginning of the string;
+                                     : -1;  // match anywhere in the string
+    return static_cast<bool>(prog.find(thread_idx, d_str, begin, end));
   }
 };
 
-struct contains_dispatch_fn {
-  reprog_device d_prog;
-  bool const beginning_only;
+std::unique_ptr<column> contains_util(strings_column_view const& input,
```

Comment:
This is used by `contains_re` and `matches_re` so I did not want to name it just `contains`.

Reply:
That makes sense, perhaps a different common regex term like `find` or `search`? Or maybe just a different suffix `contains_internal`? I just don't think `util` in particular is appropriate here, even `impl` would be a more accurate description of what this function is.

Reply:
Great meme :1st_place_medal: Sadly, a great meme does not help to convince me that the naming here makes sense. 

Right now, `contains_util` is to `contains` basically what our detail APIs are to our public APIs, except with the `beginning_only` parameter instead of the `stream`. `contains_re` and `matches_re` delegate all their logic to this internal function. `util` implies a much smaller scope than that to me.

I don't want to block the PR over this, though, so I'll let @hyperbolic2346 weigh in with his thoughts. If he doesn't have an issue here then we can move forward, otherwise we can try to brainstorm a happy medium together.

Reply:
I'm not so strongly opinionated that I would build a meme for my argument. I'm fine with leaving this as-is.